Sources/Logging/Strings/StoreKitStrings.swift

```diff
@@ -71,6 +73,10 @@ extension StoreKitStrings: CustomStringConvertible {
         case .sk1_no_known_product_type:
             return "This StoreProduct represents an SK1 product, the type of product cannot be determined, " +
             "the value will be undefined. Use `StoreProduct.productCategory` instead."
+
+        case .sk1_discount_missing_locale:
+            return "There is a problem with this SKProductDiscount missing a Locale - " +
+            "this is an issue with the App Store."
```

Comment:
"This is an issue with the App Store" is there anything we can communicate around what the developer can do in this case? cc @codykerns  for visibility

Reply:
I'm waiting to [hear back from the user regarding why they think it might be `nil`](https://github.com/RevenueCat/purchases-ios/issues/1521#issuecomment-1111186139). Once we get more context maybe we can update this.

Reply:
The user sent us some screenshots and I can't think of anything wrong in his set up. My only guess is a StoreKit bug related to the CZK App Store leading to a `nil` `NSLocale`. So @taquitos to your point, I can't think of anything the developer can do in this case, so we might just be able to communicate "hey this isn't our bug, sorry".
Maybe I'll change this to say "We will default to the device's current Locale".
What do you think /cc @aboedo?

Reply:
Historically we've gone with stuff like "this is an App Store quirk". I don't think there's much that we can do - defaulting to another Locale could be dangerous, like, we could end up showing the wrong prices for a product, and that's kind of worse than just erroring out or even crashing. 
In this particular case, Apple is breaking the nullability contract, so I don't imagine it's tied to any specific setup on the dev side. 

Hopefully SK2 will reduce the number of times we run into stuff like this given it's built in Swift

Reply:
> defaulting to another Locale could be dangerous, like, we could end up showing the wrong prices for a product, and that's kind of worse than just erroring out or even crashing.

That's a good point. That's what this PR is doing though. We could make it ignore the discount, but then this user's discount won't work at all :/

Reply:
@joshdholtz @aboedo what do you guys think? We have no more info about what could cause this so I don't know what else we could add.

Reply:
@NachoSoto I'm out of ideas what could cause this issue. I really don't think its anything we or the developer could do here. I think this is the best way we can handle for now 😇